### ISSUE TITLE
Move maximumSpecCallbackDepth as a constant on top

### ISSUE THREAD

**Issue by mtomov** [jasmine/jasmine#366]
I found it very unfortunate that this "constant" is tucked into the middle of the code in version 2.0-alpha, and I was looking for two days why I can't run more than 25 specs in a single run. 

Could you move it up to the other "constants" on top?

Thank you.

**Comment by wejrowski** [comment]
I'm running into an issue with this where we are only getting 10-20 specs running. I get a report back with  9 passed, 1 pending, 17 total

**Comment by NameFILIP** [comment]
I've also spent a couple of days chasing an issue with `maximumSpecCallbackDepth`. It either should be configurable or at least, when the `currentSpecCallbackDepth`  is getting reset to 0 (https://github.com/jasmine/jasmine/blob/141fd910f27eeecd27f66646efe00f7512649bd2/lib/jasmine-core/jasmine.js#L657) the user should be notified about this (for example: "Please increase the `maximumSpecCallbackDepth` or split your test on smaller chunks). Otherwise, somebody else will spend more days on debugging this issue.

**Comment by NameFILIP** [comment]
@slackersoft, you seem active in this repo.. could you please reopen this issue?

**Comment by slackersoft** [comment]
Tweaking the `maximumSpecCallbackDepth` should just be an optimization for your suite. It really only exists so jasmine can try to call your specs in a tail-call fashion for some time before using `setTimeout` to break the stack so we don't get a stack overflow. This means that for any suite with more than 20 specs, you should see the `currentSpecCallbackDepth` reset at least once as we use `setTimeout` to get a fresh stack.

If you're actually seeing stack overflow errors while running your jasmine suite, you would actually want to _lower_ your `maximumSpecCallbackDepth` so the stack gets reset more often.

@wejrowski or @NameFILIP can you provide a sample suite that exhibits the behavior you're seeing so we can look into what is happening?

**Comment by FuzzySockets** [comment]
Please reference my comment in https://github.com/jasmine/jasmine/issues/945. This is causing my tests to do a full page reload (using karma) once the maximumSpecCallback depth is reached. The spec causing an issue is laid out as follows:

-fdescribe
  -beforeEach
  -beforeEach
  -describe
     -beforeEach
     -afterEach
     -describe
        -it
     -describe
        -it
     -describe
        -it
        -it
        -it
        -it
        -it
        -it
     -describe
        -it
        -it
     -describe
        -it
        -it

If I delete one of the it blocks, the suite will run fine.  Notice, the number of its, describes, fdescribe = 19.

**Comment by vileppanen** [comment]
I think I'm dealing with the same issue here. But in my case, the full page reload seems to come out of my angular application run block, where I redirect to login page, if certain cookie does not yet exist.
I'm running the tests with karma 1.3.0, jasmine-core 2.4.1

At first I had an Angular (v.1.5.3) app that was bootstrapped automatically (ng-app directive in the html). Tests ran fine and everything was right. But then a business requirement had me refactor the application so, that it was bootstrapped manually --> unit tests started failing due to full page reload errors. 

I managed to drill down the issue and verify that if I have 19 describe blocks in a test suite, everything runs fine. If I have 20 describe blocks, I get full page reload error.

I did some console logging, and it seems that the applications run block gets called when the 20th describe block kicks in (this  probably relates to the stack reset). And this run block now executes the cookie/login check with all the redirects and stuff (hence the full page reload). But why the same run block is not called when I have 19 describe blocks, and how can I mock the run block Login service? And furthermore, why all the sudden these errors start appearing with manually bootstrapped application, and not in the automatically bootstrapped version?

And to confuse even more: I have the main angular application in separate module, and services I try to unit test are in completely different module. I don't even load the main application module in the unit tests.
